README.rst

```diff
@@ -41,6 +41,15 @@ Optional:
 
 -  docutils (for documentation)
 
+Platform-Specific Instructions
+""""""""""""""""""""""""""""""
+
+The build system will try to pull in as many missing dependencies as it can into the build environment. However, this has a few caveats.
+
+On **Linux**, you should use the libraries provided by your distribution's package manager (i.e: apt, yum, etc.). In particular, installing packages like ``libsdl2-dev`` is necessary, especially if your system uses Wayland, since it will pull in the necessary libraries to build for your system. Ensure that you've correctly installed the dependency packages if you run into errors like ``SDL_Error: No available video device``.
```

Comment:
I don't think Wayland is worth mentioning in the system-SDL case; the wayland headers aren't actually required for that (but the runtime libs may be). Maybe it should just recommend to use system libraries with `meson --wrap-mode=nofallback`, unless the user knows what they're doing.

The real caveat with SDL on Linux is that if you build it as a subproject, it will attempt to include support for as much stuff as possible. We can't really make missing libraries an error there, because pretty much everything is optional, and wrapping all of them in subprojects would be a maintenance nightmare.

I'm not sure how to handle this in the readme honestly, perhaps simply listing which libraries enable which functionality in SDL.